src/fs.c

```diff
@@ -58,7 +58,7 @@ static int duv_string_to_flags(duk_context *ctx, const char* string) {
   bool read = false;
   bool write = false;
   int flags = 0;
-  while (string) {
+  while (string && *string != NULL) {
```

Comment:
I see this was a typo on my part.  I'd rather change this to simply `while (string[0]) {` if you don't mind.  This function assumes string is not NULL since it's always coming from js via `duk_require_string`.